src/index/index.c

```diff
@@ -1,173 +1,316 @@
-
-  while(true) {
-    bool depleted = false;
-    GxB_MatrixTupleIter_next(it, NULL, &node_id, &depleted);
-    if(depleted) break;
-    Graph_GetNode(g, node_id, &node);
-    // If the sought property is at a different offset than it occupied in the previous node,
-    // then seek and update
-    if (attr_id != ENTITY_PROPS(&node)[prop_index].id) {
-      found = 0;
-      for (int i = 0; i < ENTITY_PROP_COUNT(&node); i ++) {
-        prop = ENTITY_PROPS(&node) + i;
-        if (attr_id == prop->id) {
-          prop_index = i;
-          found = 1;
-          break;
+// Removes fields from index.
+void Index_RemoveField
+(
+    Index *idx,
+    const char *field
+)
+{
+    assert(idx);
+    if(!Index_ContainsField(idx, field)) return;
+
+    for(uint i = 0; i < idx->fields_count; i++) {
+        if(idx->fields[i] == field) {
+            idx->fields_count--;
+            rm_free(idx->fields[i]);
+            array_del_fast(idx->fields, i);
+            array_del_fast(idx->fields_ids, i);
+            break;
         }
-      }
-    } else {
-      found = 1;
     }
-    // The targeted property does not exist on this node
-    if (!found) continue;
+}
 
-    prop = ENTITY_PROPS(&node) + prop_index;
-    // This value will be cloned within the skiplistInsert routine if necessary
-    SIValue *key = &prop->value;
+void Index_IndexNode
+(
+    Index *idx,
+    Node *n
+)
+{
+    double score = 0;           // Default score.
+    const char* lang = NULL;    // Default language.
+    RSIndex *rsIdx = idx->idx;
+    NodeID node_id = ENTITY_GET_ID(n);
 
-    sl = _select_skiplist(index, key->type);
-    if (!sl) continue; // Value was of a type not supported by indices.
-    skiplistInsert(sl, key, node_id);
-  }
+    // Create a document out of node.
+    RSDoc *doc = RediSearch_CreateDocument(&node_id, sizeof(EntityID), score, lang);
 
-  GxB_MatrixTupleIter_free(it);
+    // Add document field for each indexed property.
+    for(uint i = 0; i < idx->fields_count; i++) {
+        if(idx->fields_ids[i] == ATTRIBUTE_NOTFOUND) continue;
 
-  return index;
-}
+        SIValue *v = GraphEntity_GetProperty((GraphEntity*)n, idx->fields_ids[i]);
+        if(v == PROPERTY_NOTFOUND) continue;
+        if(idx->type == IDX_FULLTEXT) {
+            // Value must be of type string.
+            if(v->type != T_STRING) continue;
 
-//------------------------------------------------------------------------------
-// Index updates
-//------------------------------------------------------------------------------
+            RediSearch_DocumentAddFieldString(doc,
+                                            idx->fields[i],
+                                            v->stringval,
+                                            strlen(v->stringval),
+                                            RSFLDTYPE_FULLTEXT);
+        } else {
+            char buff[1024];
+            if(v->type == T_STRING) {
+                sprintf(buff, "%s_tag", idx->fields[i]);
```

Comment:
Buffer overflow risk; we should size `buff` according to `strlen(idx->fields[i])`.
(Same for 174, 226, 228).

Reply:
Removed, using RediSearch multi type field.